crates/ruff/src/version.rs

```diff
@@ -68,6 +68,11 @@ pub(crate) fn version() -> VersionInfo {
     }
 }
 
+/// Used for testing the --version command
+pub fn version_string() -> String {
+    format!("{}", version())
+}
```

Comment:
I would duplicate this information in the test instead of making the module public. Especially beause the `version_string` function isn't used in the implementation code.

Reply:
I wondered about doing this (diff is relative to this PR branch):

```diff
diff --git a/crates/ruff/src/version.rs b/crates/ruff/src/version.rs
index 6de4b2534..194dd62c7 100644
--- a/crates/ruff/src/version.rs
+++ b/crates/ruff/src/version.rs
@@ -14,7 +14,7 @@ pub(crate) struct CommitInfo {
 
 /// Ruff's version.
 #[derive(Serialize)]
-pub(crate) struct VersionInfo {
+pub struct VersionInfo {
     /// Ruff's version, such as "0.5.1"
     version: String,
     /// Information about the git commit we may have been built from.
@@ -40,7 +40,7 @@ impl fmt::Display for VersionInfo {
 }
 
 /// Returns information about Ruff's version.
-pub(crate) fn version() -> VersionInfo {
+pub fn version() -> VersionInfo {
     // Environment variables are only read at compile-time
     macro_rules! option_env_str {
         ($name:expr) => {
@@ -68,11 +68,6 @@ pub(crate) fn version() -> VersionInfo {
     }
 }
 
-/// Used for testing the --version command
-pub fn version_string() -> String {
-    format!("{}", version())
-}
-
 #[cfg(test)]
 mod tests {
     use insta::{assert_display_snapshot, assert_json_snapshot};
diff --git a/crates/ruff/tests/version.rs b/crates/ruff/tests/version.rs
index 69e8ea1ca..360747722 100644
--- a/crates/ruff/tests/version.rs
+++ b/crates/ruff/tests/version.rs
@@ -7,10 +7,12 @@ use insta_cmd::{assert_cmd_snapshot, get_cargo_bin};
 use regex::escape;
 use tempfile::TempDir;
 
-use ruff::version::version_string;
-
 const BIN_NAME: &str = "ruff";
 
+fn version_string() -> String {
+    format!("{}", ruff::version::version())
+}
+
 #[test]
```

But that actually exposes more of the module publicly rather than less (which is why I went with what I have currently)

Reply:
Sorry yeah, see my other comment (I deleted it but didn't realise that I submitted it right away). I recommend using filters here.